### PR TITLE
Consolidate the storage implementations + Remove libcloud dependency

### LINKED ISSUE #618: Storage: Prevent leaking storage backend connections
[Project board link](https://github.com/orgs/k8ssandra/projects/8/views/1?pane=issue&itemId=36147728)

Various storage implementations do not handle connections to the backend correctly. With libcloud, we did care about closing connections.

As we replaced libcloud with the aiohttp-s3-client, it became obvious the connections are leaking. A naive implementation of placing the closing logic in a __del__ method did not work. When running ITs, a fast succession of steps would just call several __init__s of the storage implementation, but somehow forget about the __del__s.

We'll need to spend more time doing this properly. One approach is to use the `with` keyword any time we instantiate a Storage. This way we'll have more control, because calls to __enter__ and __exit__ are more reliable. 

# Definition of done
- [ ] S3 (and compatible) storage backed does not leak connections
- [ ] GCS storage backend does not leak connections
- [ ] Azure storage backend does not leak conenctions

### PR DESCRIPTION
**Pull request by rzvoncek**
Fixes #618 .

### CODE REVIEW

**Comment by rzvoncek** [comment]
Also pushed a change to fix the bug reported by SonarCloud.

File: k8s/Dockerfile-azure [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM ubuntu:18.04 as base
+k8s/Dockerfile-azure FROM ubuntu:18.04 as base
```
Comment (rzvoncek) [inline k8s/Dockerfile-azure:1]: I'm actally not sure how this sneakedi n.

Comment (adejanovski) [reply k8s/Dockerfile-azure:1]: lol, we're removing this file anyway.

File: medusa/storage/__init__.py [hunk 2]
```diff
@@ -72,39 +66,36 @@ def __init__(self, *, config):
...
+        self.storage_driver.connect()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.storage_driver.disconnect()
+
+    def _load_storage(self):
         logging.debug('Loading storage_provider: {}'.format(self._config.storage_provider))
-        if self._config.storage_provider == Provider.GOOGLE_STORAGE:
+        if self._config.storage_provider.lower() == 'google_storage':
```
Comment (adejanovski) [inline medusa/storage/__init__.py:81]: **Suggestion:** we should create a refactoring ticket to create new constants for the storage providers.

File: medusa/storage/azure_storage.py [hunk 3]
```diff
@@ -306,31 +145,76 @@ async def _download_blob(self, src: str, dest: str):
...
 
-    def get_cache_path(self, path: str) -> str:
-        logging.debug(
-            "My cache path is {}".format(path)
+        with open(src, "rb") as data:
+            blob_client = await self.azure_container_client.upload_blob(
+                name=object_key,
+                data=data,
+                overwrite=True,
+                max_concurrency=4,
```
Comment (adejanovski) [inline medusa/storage/azure_storage.py:187]: **Suggestion:** we had settled on 16 ;)

**Review by adejanovski — APPROVED** [review]
Fantastic job!
We're good for a merge (with two refactoring tickets that need to be created)

File: medusa/storage/__init__.py (hunk 2 above)
Comment (rzvoncek) [reply medusa/storage/__init__.py:81]: https://github.com/thelastpickle/cassandra-medusa/issues/646

File: medusa/storage/azure_storage.py (hunk 3 above)
Comment (rzvoncek) [reply medusa/storage/azure_storage.py]: Fixed.

**Pull request merged**